web/src/buddy_list.js

```diff
@@ -27,8 +49,18 @@ class BuddyListConf {
 
     get_li_from_user_id(opts) {
         const user_id = opts.user_id;
-        const $container = $(this.container_selector);
-        return $container.find(`${this.item_selector}[data-user-id='${CSS.escape(user_id)}']`);
+        const $users_matching_view_container = $(this.matching_view_list_selector);
+        const $li = $users_matching_view_container.find(
+            `${this.item_selector}[data-user-id='${CSS.escape(user_id)}']`,
+        );
+        if ($li.length > 0) {
+            return $li;
+        }
+
+        const $other_users_container = $(this.other_user_list_selector);
+        return $other_users_container.find(
+            `${this.item_selector}[data-user-id='${CSS.escape(user_id)}']`,
+        );
```

Comment:
This code seems unnecessarily duplicative; couldn't this just use a container containing both elements with a single `$container.find(\`${this.item_selector}[data-user-id='${CSS.escape(user_id)}']\`)`?

Reply:
Oh true! Thanks :)